### PR TITLE
Use the new Package Manager wrapper in the CLI

### PR DESCRIPTION
**Pull request by costinsin**
## Type of change

-   [x] 🍕 New feature

## Description

Replaces all the old usages of the `npm install` with the new package manager wrapper.

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

**Comment by costinsin** [comment]
Current dependencies on/for this PR:
* dev
  * **PR #559** <a href="https://app.graphite.dev/github/pr/Genez-io/genezio/559?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #560** <a href="https://app.graphite.dev/github/pr/Genez-io/genezio/560?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Genez-io/genezio/560?utm_source=stack-comment).

File: src/genezio.ts [hunk 1]
```diff
@@ -42,6 +45,17 @@ if (ENABLE_DEBUG_LOGS_BY_DEFAULT) {
...
+// setup package manager
+const configuration = await getProjectConfiguration();
+if (configuration.packageManager) {
+    if (configuration.packageManager in Object.keys(PackageManagerType)) {
+        setPackageManager(configuration.packageManager);
+    } else {
+        log.warn(`Unknown package manager '${configuration.packageManager}'. Using 'npm' instead.`);
+        setPackageManager(PackageManagerType.npm);
+    }
+}
```
Comment (costinsin) [inline src/genezio.ts:57]: @vladiulianbogdan I am not sure if this code should stay here or be moved in `src/packageManagers/packageManager.ts`. The latter option has the advantage that all code related to the package manages stays together. What do you think?

Comment (vladiulianbogdan) [reply src/genezio.ts:57]: I think it is better here. If we move this to the `packageManager.ts`, then we pollute a class with yaml and project logic. This file is the one that does the wiring, so to respect the single responsibility principle, I think it is better to have this logic here.

**Review by vladiulianbogdan — APPROVED** [review]

**Comment by costinsin** [comment]
### Merge activity

* **Nov 22, 11:04 AM**: @costinsin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Genez-io/genezio/560).

**Pull request merged**